hazelcast-client/src/main/java/com/hazelcast/client/spi/impl/discovery/HazelcastCloudDiscovery.java

```diff
@@ -0,0 +1,135 @@
+/*
+ * Copyright (c) 2008-2018, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.client.spi.impl.discovery;
+
+import com.eclipsesource.json.Json;
+import com.eclipsesource.json.JsonValue;
+import com.hazelcast.nio.Address;
+import com.hazelcast.util.StringUtil;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.net.HttpURLConnection;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.net.URL;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Scanner;
+
+import static com.hazelcast.util.ExceptionUtil.rethrow;
+
+/**
+ * Discovery service that discover nodes via hazelcast.cloud
+ * https://coordinator.hazelcast.cloud/cluster/discovery?token=<TOKEN>
+ */
+class HazelcastCloudDiscovery {
+
+    private static final int DEFAULT_TIMEOUT_MILLIS = 30000;
```

Comment:
should this be related to the ClientNetworkConfig.setConnectionTimeout(int connectionTimeout) ? can it be greater than this config?

Reply:
Make sense. I will reuse `ClientNetworkConfig.setConnectionTimeout` here